components/provisioner/internal/parallel-installation/components_status.go

```diff
@@ -0,0 +1,83 @@
+package parallel_installation
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/control-plane/components/provisioner/internal/model"
+
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/components"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/deployment"
+)
+
+type ComponentsStatus struct {
+	components         map[string]bool
+	componentFailed    error
+	installationFailed error
+}
+
+func NewComponentsStatus(components []model.KymaComponentConfig) *ComponentsStatus {
+	cmpList := make(map[string]bool)
+	for _, component := range components {
+		cmpKey := componentKey(string(component.Component), component.Namespace)
+		cmpList[cmpKey] = false
+	}
+
+	return &ComponentsStatus{components: cmpList}
+}
+
+func (cs *ComponentsStatus) ConsumeEvent(event deployment.ProcessUpdate) {
+	// check if installation process failed
+	if event.Event == deployment.ProcessExecutionFailure ||
+		event.Event == deployment.ProcessForceQuitFailure ||
+		event.Event == deployment.ProcessTimeoutFailure {
+		cs.installationFailed = fmt.Errorf("status: %s, error: %v", event.Event, event.Error)
+	}
+
+	// check if component installation process succeeded
+	key := componentKey(event.Component.Name, event.Component.Namespace)
+	if _, ok := cs.components[key]; !ok {
```

Comment:
Another crash observed during stress tests:
```
goroutine 6737 [running]:
github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation.(*ComponentsStatus).ConsumeEvent(0x0, 0x239565b, 0xe, 0x239ac17, 0x11, 0x0, 0x0, 0xc004a17110, 0x15, 0xc000b9ba00, ...)
	/go/src/github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation/components_status.go:38 +0x152
github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation.parallelInstallationService.getCallbackUpdate.func2(0x239565b, 0xe, 0x239ac17, 0x11, 0x0, 0x0, 0xc004a17110, 0x15, 0xc000b9ba00, 0x10, ...)
	/go/src/github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation/parallel_installation.go:84 +0xbc
github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation.parallelInstallationService.getCallbackUpdate.func3(0x239565b, 0xe, 0x239ac17, 0x11, 0x0, 0x0, 0xc004a17110, 0x15, 0xc000b9ba00, 0x10, ...)
	/go/src/github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation/parallel_installation.go:95 +0x335
github.com/kyma-incubator/hydroform/parallel-install/pkg/deployment.(*core).processUpdateComponent(0xc002690810, 0x239ac17, 0x11, 0xc004a17110, 0x15, 0xc000b9ba00, 0x10, 0xc000b9b100, 0xa, 0x238ceea, ...)
	/go/pkg/mod/github.com/kyma-incubator/hydroform/parallel-install@v0.0.0-20210610074205-bbaf6958f1a5/pkg/deployment/core.go:115 +0x155
github.com/kyma-incubator/hydroform/parallel-install/pkg/deployment.(*Deployment).deployComponents(0xc0022681a8, 0x26af3c0, 0xc0071455c0, 0xc007354840, 0x239ac17, 0x11, 0xc007145580, 0x1174f2ca8ea, 0x15d289160ea, 0x0, ...)
	/go/pkg/mod/github.com/kyma-incubator/hydroform/parallel-install@v0.0.0-20210610074205-bbaf6958f1a5/pkg/deployment/deployment.go:153 +0x358
github.com/kyma-incubator/hydroform/parallel-install/pkg/deployment.(*Deployment).startKymaDeployment(0xc0022681a8, 0x2677b78, 0xc007145500, 0xc007145540, 0xc007145580, 0x0, 0x0)
	/go/pkg/mod/github.com/kyma-incubator/hydroform/parallel-install@v0.0.0-20210610074205-bbaf6958f1a5/pkg/deployment/deployment.go:130 +0x5d8
github.com/kyma-incubator/hydroform/parallel-install/pkg/deployment.(*Deployment).StartKymaDeployment(0xc0022681a8, 0xc006652240, 0xc001987a90)
	/go/pkg/mod/github.com/kyma-incubator/hydroform/parallel-install@v0.0.0-20210610074205-bbaf6958f1a5/pkg/deployment/deployment.go:85 +0x2d6
github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation.(*Deployer).StartKymaDeployment(0xc00263a1e0, 0x0, 0x0)
	/go/src/github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation/deployer.go:24 +0x7a
github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation.AsyncDeployment.StartKymaDeployment.func1(0x2661c60, 0xc00263a1e0, 0x2494ff8, 0x2495000)
	/go/src/github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation/parallel_installation.go:40 +0x31
created by github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation.AsyncDeployment.StartKymaDeployment
	/go/src/github.com/kyma-project/control-plane/components/provisioner/internal/parallel-installation/parallel_installation.go:39 +0x5d
```

At the first glance I cannot tell what was the cause here.

Reply:
It looks similar to [previous](https://github.com/kyma-project/control-plane/pull/678#discussion_r653338558) issue (lines are the same).